packages/api/src/middlewares/RouterMiddlewareFactory.ts

```diff
@@ -1,32 +1,33 @@
 import { asValue, AwilixContainer } from "awilix";
 import { Context, Middleware } from "koa";
-import { asClassMethod } from "../AwilixHelpers";
-import { RequestParsingService } from "../config/RequestParsingService";
-import { RouteTransformationService } from "../config/RouteTransformationService";
 import { getMethod, HttpMethod } from "../HttpMethod";
-import { IRouter } from "../reflection/IRouterClass";
+import { MethodNotSupportedError } from "../web/MethodNotSupportedError";
 import { RouteRegistry } from "../web/RouteRegistry";
+import { UnknownMethodError } from "../web/UnknownMethodError";
+import { UnknownRouteError } from "../web/UnknownRouteError";
 import { IMiddlewareFactory } from "./IMiddlewareFactory";
 import { INextCallback } from "./INextCallback";
 
+/**
+ * Middleware factory that provides a routing middleware. The created
+ * middleware will examine incoming requests, map them to the proper route
+ * handler, create a request-scoped DI container, and use it to invoke the
```

Comment:
Update docs. Doesn't create the requestContainer itself anymore. Actually relies on it being already provided in ctx.state.requestContainer.